### PR TITLE
rework background and textstyle classes

### PR DESCRIPTION
**Pull request by Alex009**
#138 separate solid and Gradient background in api to limit where used sublayer background in iOS
#13 add support of different textcolor for button states
#117 remove text-specific code - wrapper of UILabel is removed

also removed FlatAlertWidget - it can be replaced by combination of other widgets with more customizations

### CODE REVIEW

File: widgets/src/commonMain/kotlin/dev/icerock/moko/widgets/style/state/FocusableState.kt [hunk 1]
```diff
@@ -0,0 +1,12 @@
+/*
+ * Copyright 2020 IceRock MAG Inc. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package dev.icerock.moko.widgets.style.state
+
+data class FocusableState<out T>(
```
Comment (Alex009) [inline widgets/src/commonMain/kotlin/dev/icerock/moko/widgets/style/state/FocusableState.kt]: must be used in factories with focusable colors

File: widgets/src/commonMain/kotlin/dev/icerock/moko/widgets/style/state/SelectableState.kt [hunk 2]
```diff
@@ -0,0 +1,12 @@
+/*
+ * Copyright 2020 IceRock MAG Inc. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package dev.icerock.moko.widgets.style.state
+
+data class SelectableState<out T>(
```
Comment (Alex009) [inline widgets/src/commonMain/kotlin/dev/icerock/moko/widgets/style/state/SelectableState.kt]: must be used in bottomnavitem colors

File: widgets/src/iosMain/kotlin/dev/icerock/moko/widgets/utils/BackgroundExt.kt [hunk 3]
```diff
@@ -165,10 +153,10 @@ fun UIView.applyBackgroundIfNeeded(background: Background?) {
...
-
-    // TODO complete it
-    applyBackgroundIfNeeded(stateBackground.normal)
-}
+//fun UIControl.applyStateBackgroundIfNeeded(stateBackground: PressableState<Background<out Fill>>?) {
+//    if (stateBackground == null) return
+//
+//    // TODO complete it
+//    applyBackgroundIfNeeded(stateBackground.normal)
+//}
```
Comment (Alex009) [inline widgets/src/iosMain/kotlin/dev/icerock/moko/widgets/utils/BackgroundExt.kt]: delete it

File: widgets/src/commonMain/kotlin/dev/icerock/moko/widgets/factory/ButtonWithIconViewFactory.kt [hunk 4]
```diff
@@ -4,27 +4,28 @@
...
+    background: PressableState<Background<out Fill>>? = null,
+    textStyle: TextStyle<PressableState<Color>>? = null,
     isAllCaps: Boolean? = null,
     padding: PaddingValues? = null,
     margins: MarginValues? = null,
     androidElevationEnabled: Boolean? = null,
     iconGravity: IconGravity? = null,
     iconPadding: Float? = null,
-    icon: ImageResource
+    icon: PressableState<ImageResource>
```
Comment (kovalandrew) [inline widgets/src/commonMain/kotlin/dev/icerock/moko/widgets/factory/ButtonWithIconViewFactory.kt]: Field names of PressableStates not clearly understand. For parameters `background` and `icon` required type `PressableState`

Comment (kovalandrew) [inline widgets/src/commonMain/kotlin/dev/icerock/moko/widgets/factory/ButtonWithIconViewFactory.kt]: Field names of PressableStates not clearly understand. For parameters background and icon required type PressableState

Comment (Alex009) [reply widgets/src/commonMain/kotlin/dev/icerock/moko/widgets/factory/ButtonWithIconViewFactory.kt]: i think name should not contain type info. name should be clear and show what data here, type should show type :) mix of type and name in strong-typed language is bad i think.

for example:
`background: PressableState<Background<out Fill>>?`
will be 
`stateBackground: PressableState<Background<out Fill>>?`
or
`backgroundStates: PressableState<Background<out Fill>>?`

IDE and compiler can suggest and check types for us :)

Comment (kovalandrew) [reply widgets/src/commonMain/kotlin/dev/icerock/moko/widgets/factory/ButtonWithIconViewFactory.kt]: Ok, let's see it on practice)

**Pull request merged**